Mixin: added MimirKafkaClientBufferedProduceBytesTooHigh alert and 'Kafka produced records / sec' panel (#8763)
grafana/mimir: docs/sources/mimir/manage/mimir-runbooks/_index.md

- Check if ingesters are processing too many records, and they need to be scaled up (vertically or horizontally).
- Check actual error in logs to see whether the `-ingest-storage.kafka.wait-strong-read-consistency-timeout` or the request timeout has been hit first.

### MimirKafkaClientBufferedProduceBytesTooHigh

This alert fires when the Kafka client buffer, used to write incoming write requests to Kafka, is getting full.

> How it **works**:

Author: Skipping this suggestion, because it's consistent with the rest of the file. You can open a PR to change it everywhere if it's something worth to you.

- Distributor and ruler encapsulate write requests into Kafka records and send them to Kafka.

> - The Kafka client has a limit on the total bytes size of records buffered to be sent to Kafka or sent to Kafka but not acknowledged yet.

Reviewer: ```suggestion
- The Kafka client has a limit on the total byte size of buffered records either sent to Kafka or sent to Kafka but not acknowledged yet.
```

> - When the limit is reached, Kafka client will reject to produce more records and it will fast fail.

Reviewer: ```suggestion
- When the limit is reached, the Kafka client stops producing more records and fast fails.
```

Reviewer: We try to avoid using future tense in docs.

- The limit is configured via `-ingest-storage.kafka.producer-max-buffered-bytes`.

> - The default limit is configured intentionally high so that when the buffer utilization gets close to the limit then there's probably an issue.

Reviewer: ```suggestion
- The default limit is configured intentionally high, so that when the buffer utilization gets close to the limit, this indicates that there's probably an issue.
```

> 

Reviewer: Is this a sequential order of steps? If so, we could use an ordered list instead of bullet points.

Author: Not strictly sequential.

> How to **investigate**:

Reviewer: ```suggestion
How to investigate:
```

Reviewer: Same as above, bold is only for UI elements.

Author: _Same comment here about this being a pattern in the whole file._

- Query `cortex_ingest_storage_writer_buffered_produce_bytes{quantile="1.0"}` metrics to see the actual buffer utilization peaks.

>   - If the buffer high utilization is isolated to a small set of pods, then there may be a issue in the client pods.

Reviewer: ```suggestion
  - If the buffer high utilization is isolated to a small set of pods, then there might be an issue in the client pods.
```

Reviewer: I'm not sure "buffer high utilization" is the best wording. Could we say "high buffer utilization"?

Author: > Could we say "high buffer utilization"?

Yes, we can. Edited.

>   - If the buffer high utilization is spread across all or most pods, then there may be a issue in Kafka.

Reviewer: ```suggestion
  - If the buffer high utilization is spread across all or most pods, then there might be an issue in Kafka.
```

Reviewer: Same as above wrt "buffer high utilization".
